### PR TITLE
src: Support strikethrough faces

### PR DESCRIPTION
**Pull request by eraserhd**
ncurses does not support strikethrough (intentionally, because they ran out of bits?) but Kitty does (at least), and the code reserved for it is quite old, so it should be safe.

[See here](https://stackoverflow.com/questions/8357203/is-it-possible-to-display-text-in-a-console-with-a-strike-through-effect)

We could support the face but do nothing with it on `master` if you like.

### CODE REVIEW

**Comment by lenormf** [comment]
Do you have a usecase for this? Markup?

**Comment by eraserhd** [comment]
This came up working on kak-jira.  JIRA has markup for added and removed text.  These are rendered on the web as underline and strikethrough.

That being said, IMHO text rendering in Kakoune doesn’t need to match anything else — this seemed like it could be generally useful.

File: src/face.hh [hunk 1]
```diff
@@ -9,17 +9,18 @@ namespace Kakoune
...
+    Underline     = 1 << 1,
+    Reverse       = 1 << 2,
+    Blink         = 1 << 3,
+    Bold          = 1 << 4,
+    Dim           = 1 << 5,
+    Italic        = 1 << 6,
+    FinalFg       = 1 << 7,
+    FinalBg       = 1 << 8,
+    FinalAttr     = 1 << 9,
+    Strikethrough = 1 << 10,
```
Comment (mawww) [inline src/face.hh:22]: I'd rather have all non-final attributes together, so could you move it next to italic, as `1 << 7`

File: src/terminal_ui.cc [hunk 2]
```diff
@@ -181,7 +181,22 @@ void TerminalUI::Screen::set_face(const Face& face)
 {
     static constexpr int fg_table[]{ 39, 30, 31, 32, 33, 34, 35, 36, 37, 90, 91, 92, 93, 94, 95, 96, 97 };
     static constexpr int bg_table[]{ 49, 40, 41, 42, 43, 44, 45, 46, 47, 100, 101, 102, 103, 104, 105, 106, 107 };
-    static constexpr int attr_table[]{ 0, 4, 7, 5, 1, 2, 3 };
```
Comment (mawww) [inline src/terminal_ui.cc]: while the new code is a bit more explicit, I kind of like the simplicity of having a onliner table, which should be possible again if you change Strikethrough as requested above.

File: src/terminal_ui.cc [hunk 3]
```diff
@@ -199,10 +214,10 @@ void TerminalUI::Screen::set_face(const Face& face)
     bool join = false;
     if (face.attributes != m_active_face.attributes)
     {
-        for (int i = 0; i < sizeof(attr_table) / sizeof(int); ++i)
+        for (int i = 0; i < sizeof(attr_table) / sizeof(attr_table[0]); ++i)
```
Comment (mawww) [inline src/terminal_ui.cc:217]: we could use `std::size(attr_table)` here to avoid having to do the division dance.

**Review by mawww — CHANGES_REQUESTED** [review]

**Comment by eraserhd** [comment]
@mawww All requested changes made.

(I also tried to make the attribute bit numbers equal to the terminal codes, which required skipping a few, but it ended up not being an improvement.)

**Comment by mawww** [comment]
cherry-picked

**Pull request closed without merging**